src/Control/Carrier/DockerEngineApi.hs

```diff
@@ -0,0 +1,104 @@
+import Control.Effect.Lift (Lift, sendIO)
+import Data.Aeson (FromJSON (parseJSON), eitherDecode, withObject, (.:))
+import Data.Conduit.Binary (sinkFile)
+import Data.String.Conversion (toString, toText)
+import Data.Text (Text)
+import Network.HTTP.Client qualified as HTTP
+import Network.HTTP.Client.Internal (Connection)
+import Network.HTTP.Conduit qualified as HTTPConduit
+import Network.HTTP.Types (Status (statusCode))
+import Network.Socket qualified as Socket
+import Network.Socket.ByteString qualified as SocketByteString
+import Path (Abs, File, Path)
+
+type DockerEngineApiC = SimpleC DockerEngineApiF
+
+runDockerEngineApi :: (Has (Lift IO) sig m, Has Diag.Diagnostics sig m) => DockerEngineApiC m a -> m a
+runDockerEngineApi = interpret $ \case
+  ExportImage img path -> exportDockerImage img path
+  GetImageSize img -> getDockerImageSize img
+  IsDockerEngineAccessible -> isDockerEngineAccessible
+
+-- | Exports Docker Image to a given path.
+-- Refer to: https://docs.docker.com/engine/api/v1.28/#tag/Image/operation/ImageGet
+exportDockerImage :: (Has Diag.Diagnostics sig m, Has (Lift IO) sig m) => Text -> Path Abs File -> m ()
+exportDockerImage img sinkTarget = do
+  let request = HTTP.parseRequest_ $ baseApi <> "images/" <> (toString img) <> "/get"
+  manager <- sendIO dockerClient
+  -- Performs equivalent of for given image:
+  -- >> curl --unix-socket /var/run/docker.sock -X GET "http://localhost/v1.28/images/redis:alpine/get" > img.tar
+  sendIO . runResourceT $ do
+    response <- HTTPConduit.http request manager
+    runConduit $ HTTPConduit.responseBody response .| sinkFile (toString sinkTarget)
+
+-- | Gets Image Size from Image Description Json
+-- Refer to: https://docs.docker.com/engine/api/v1.28/#tag/Image/operation/ImageInspect
+getDockerImageSize :: (Has Diag.Diagnostics sig m, Has (Lift IO) sig m) => Text -> m Int
+getDockerImageSize img = do
+  let request = HTTP.parseRequest_ $ baseApi <> "images/" <> (toString img) <> "/json"
+
+  -- Performs equivalent of for given image:
+  -- >> curl --unix-socket /var/run/docker.sock -X GET "http://localhost/v1.28/images/redis:alpine/json"
+  response <- sendIO $ HTTP.httpLbs request =<< dockerClient
+
+  let body = HTTP.responseBody response
+  case eitherDecode body of
+    Left err -> fatalText . toText $ err
+    Right (DockerImageInspectJson imgSize) -> pure imgSize
+
+-- | True if Docker Engine API is accessible, otherwise False.
+-- Refer to: https://docs.docker.com/engine/api/v1.28/#tag/System/operation/SystemPing
+isDockerEngineAccessible :: (Has Diag.Diagnostics sig m, Has (Lift IO) sig m) => m Bool
+isDockerEngineAccessible = do
+  -- Performs equivalent of for given image:
+  -- >> curl --unix-socket /var/run/docker.sock -X GET "http://localhost/v1.28/_ping"
+  response <- sendIO $ HTTP.httpLbs (HTTP.parseRequest_ $ baseApi <> "_ping") =<< dockerClient
+  pure $ statusCode (HTTP.responseStatus response) == 200
```

Comment:
You can compare these direction with something like:
`HTPP.responseStatus response == Network.HTTP.Types.ok200`